lib/helpers.rb

```diff
@@ -54,6 +54,8 @@ def vote_for(obj)
 
   def flag_as_abuse(obj)
     raise ArgumentError, t(:user_id_is_required) unless user
+    # If this is the first abuse flag, count it, otherwise don't increment the count
+    obj.author.update_stats_for_course(obj.course_id, active_flags: 1) if obj.abuse_flaggers.empty?
```

Comment:
Sorry for not noticing this earlier, but I think there's a subtle bug here -- if stats for this course don't exist yet, `update_stats_for_course()` will automatically generate them from existing data, but when doing that, it will not see the update to `obj.abuse_flaggers`.

So in case:
1. course stats don't yet exist; and
2. this is the first abuse flag for this object

Then course stats will be automatically generated from existing data and will not see the abuse flag on this comment.

Reply:
A similar thing can happen inside the `un_flag_as_abuse` method as well.

Reply:
Ah! That explains it! I was looking into this exact issue, but didn't have much time to work on friday. I'll look into a fix for this.